flight/PiOS/Common/pios_hal.c

```diff
@@ -0,0 +1,748 @@
+					PIOS_Assert(0);
+				}
+
+				PIOS_HAL_SetReceiver(MANUALCONTROLSETTINGS_CHANNELGROUPS_PPM, ppm_rcvr_id);
+			}
+#endif  /* PIOS_INCLUDE_PPM */
+			break;
+
+		case HWSHARED_PORTTYPES_DISABLED:
+			break;
+		case HWSHARED_PORTTYPES_TELEMETRY:
+			PIOS_HAL_ConfigureCom(usart_port_cfg, PIOS_COM_TELEM_RF_RX_BUF_LEN, PIOS_COM_TELEM_RF_TX_BUF_LEN, com_driver, &port_driver_id);
+			target = &pios_com_telem_serial_id;
+			break;
+		case HWSHARED_PORTTYPES_GPS:
+			PIOS_HAL_ConfigureCom(usart_port_cfg, PIOS_COM_GPS_RX_BUF_LEN, PIOS_COM_GPS_TX_BUF_LEN, com_driver, &port_driver_id);
+			target = &pios_com_gps_id;
+			break;
+		case HWSHARED_PORTTYPES_DSM:
+#if defined(PIOS_INCLUDE_DSM)
+			if (dsm_cfg && usart_dsm_hsum_cfg) {
+				PIOS_HAL_ConfigureDSM(usart_dsm_hsum_cfg, dsm_cfg, com_driver, dsm_bind);
+			}
+#endif  /* PIOS_INCLUDE_DSM */
+			break;
+		case HWSHARED_PORTTYPES_SBUS:
+#if defined(PIOS_INCLUDE_SBUS) && defined(PIOS_INCLUDE_USART)
+			if (sbus_cfg && sbus_rcvr_cfg) {
+				uintptr_t usart_sbus_id;
+				if (PIOS_USART_Init(&usart_sbus_id, sbus_rcvr_cfg)) {
+					PIOS_Assert(0);
+				}
+				uintptr_t sbus_id;
+				if (PIOS_SBus_Init(&sbus_id, sbus_cfg, com_driver, usart_sbus_id)) {
+					PIOS_Assert(0);
+				}
+				uintptr_t sbus_rcvr_id;
+				if (PIOS_RCVR_Init(&sbus_rcvr_id, &pios_sbus_rcvr_driver, sbus_id)) {
+					PIOS_Assert(0);
+				}
+				PIOS_HAL_SetReceiver(MANUALCONTROLSETTINGS_CHANNELGROUPS_SBUS, sbus_rcvr_id);
+				pios_rcvr_group_map[MANUALCONTROLSETTINGS_CHANNELGROUPS_SBUS] = sbus_rcvr_id;
+			}
+#endif  /* PIOS_INCLUDE_SBUS */
+			break;
+		case HWSHARED_PORTTYPES_HOTTSUMD:
+		case HWSHARED_PORTTYPES_HOTTSUMH:
+#if defined(PIOS_INCLUDE_HSUM)
+			if (usart_dsm_hsum_cfg) { 
+				enum pios_hsum_proto proto;
+				switch (port_type) {
+					case HWSHARED_PORTTYPES_HOTTSUMD:
+						proto = PIOS_HSUM_PROTO_SUMD;
+						break;
+					case HWSHARED_PORTTYPES_HOTTSUMH:
+						proto = PIOS_HSUM_PROTO_SUMH;
+						break;
+					default:
+						PIOS_Assert(0);
+						break;
+				}
+				PIOS_HAL_ConfigureHSUM(usart_dsm_hsum_cfg, com_driver, &proto);
+			}
+#endif  /* PIOS_INCLUDE_HSUM */
+			break;
+		case HWSHARED_PORTTYPES_DEBUGCONSOLE:
+#if defined(PIOS_INCLUDE_DEBUG_CONSOLE)
+			PIOS_HAL_ConfigureCom(usart_port_cfg, 0, PIOS_COM_DEBUGCONSOLE_TX_BUF_LEN, com_driver, &port_driver_id);
+			target = &pios_com_debug_id;
+#endif  /* PIOS_INCLUDE_DEBUG_CONSOLE */
+			break;
+		case HWSHARED_PORTTYPES_COMBRIDGE:
+			PIOS_HAL_ConfigureCom(usart_port_cfg, PIOS_COM_BRIDGE_RX_BUF_LEN, PIOS_COM_BRIDGE_TX_BUF_LEN, com_driver, &port_driver_id);
+			target = &pios_com_bridge_id;
+			break;
+		case HWSHARED_PORTTYPES_MAVLINKTX:
+#if defined(PIOS_INCLUDE_MAVLINK)
+			PIOS_HAL_ConfigureCom(usart_port_cfg, 0, PIOS_COM_MAVLINK_TX_BUF_LEN, com_driver, &port_driver_id);
+			target = &pios_com_mavlink_id;
+#endif          /* PIOS_INCLUDE_MAVLINK */
+			break;
+		case HWSHARED_PORTTYPES_MAVLINKTX_GPS_RX:
+#if defined(PIOS_INCLUDE_MAVLINK)
```

Comment:
I think so. It is a more graceful way to fail.
On Aug 17, 2015 3:42 AM, "Michael" <notifications@github.com> wrote:

> In flight/PiOS/Common/pios_hal.c
> <https://github.com/TauLabs/TauLabs/pull/1759#discussion_r37167812>:
>
> > +			PIOS_HAL_ConfigureCom(usart_port_cfg, 0, PIOS_COM_DEBUGCONSOLE_TX_BUF_LEN, com_driver, &port_driver_id);
> > +			target = &pios_com_debug_id;
> > +#endif  /* PIOS_INCLUDE_DEBUG_CONSOLE */
> > +			break;
> > +		case HWSHARED_PORTTYPES_COMBRIDGE:
> > +			PIOS_HAL_ConfigureCom(usart_port_cfg, PIOS_COM_BRIDGE_RX_BUF_LEN, PIOS_COM_BRIDGE_TX_BUF_LEN, com_driver, &port_driver_id);
> > +			target = &pios_com_bridge_id;
> > +			break;
> > +		case HWSHARED_PORTTYPES_MAVLINKTX:
> > +#if defined(PIOS_INCLUDE_MAVLINK)
> > +			PIOS_HAL_ConfigureCom(usart_port_cfg, 0, PIOS_COM_MAVLINK_TX_BUF_LEN, com_driver, &port_driver_id);
> > +			target = &pios_com_mavlink_id;
> > +#endif          /* PIOS_INCLUDE_MAVLINK */
> > +			break;
> > +		case HWSHARED_PORTTYPES_MAVLINKTX_GPS_RX:
> > +#if defined(PIOS_INCLUDE_MAVLINK)
>
> Should this configure GPS rx even if MAVLINK is not included?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/TauLabs/TauLabs/pull/1759/files#r37167812>.
>